boards/arduino-atmega-common/include/periph_conf.h

```diff
@@ -139,6 +141,25 @@ extern "C" {
 #endif
 /** @} */
 
+/**
+ * @name ADC configuration
+ *
+ * The number of ADC channels of the atmega328p depends on the package:
+ *  - 6-channel 10-bit ADC in PDIP package
+ *  - 8-channel 10-bit ADC in TQFP and QFN/MLF package
+ * Arduino UNO / Duemilanove has thereby 6 channels. But only 5 channels can be
+ * used for ADC, because the pin of ADC5 emulate a software triggered interrupt.
+ *  -> boards -> arduino-atmega-common -> include -> board_common.h
+ *
+ * @{
+ */
+#if defined (CPU_ATMEGA328P)
+#define ADC_NUMOF       (5)
```

Comment:
I suggest to set it to 8. For a project I remapped the software interrupt to another pin since it's easily doable. I also think the 1281 does not have lines over 7, maybe you can turn the defines around to simplify the code
```
#ifdef 2560
16
#else
8
#endif

Reply:
@smlng what do you think about it?

Reply:
considering #7528, plus the fact that the uno has 6 ADC while the nano/mini have 8 ADC but all atmega328p based, it might be better to move this into a board specific periph_conf. h instead anyway. So @lebrush is right, i.e., setting ADC_NUMOF to 5 for 328p is not correct for all boards.